### PR TITLE
Initial pull method

### PR DESCRIPTION
**Pull request by SupremoUGH**
## Description

Adds the `initial_pull` and `dense_initial_pull` methods on the pallet. 

Goes together with:
- https://github.com/Manta-Network/manta-rs/pull/329
- https://github.com/Manta-Network/sdk/pull/108

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by ghzlatarev** [comment]
Don't try to restart single jobs, I think there's a bug somewhere and they just hang. Better restart all

File: Cargo.lock
Comment (ghzlatarev) [inline Cargo.lock:1]: this seems like too many changes

File: pallets/manta-pay/Cargo.toml [hunk 1]
```diff
@@ -107,16 +107,16 @@ sp-std = { git = 'https://github.com/paritytech/substrate.git', branch = "polkad
...
-manta-crypto = { git = "https://github.com/manta-network/manta-rs.git", tag = "v0.5.12", default-features = false }
-manta-pay = { git = "https://github.com/manta-network/manta-rs.git", tag = "v0.5.12", default-features = false, features = ["groth16", "parameters", "scale"] }
+manta-accounting = { git = "https://github.com/manta-network/manta-rs.git", branch = "signer_initial_sync_checkout", default-features = false }
+manta-crypto = { git = "https://github.com/manta-network/manta-rs.git", branch = "signer_initial_sync_checkout", default-features = false }
+manta-pay = { git = "https://github.com/manta-network/manta-rs.git", branch = "signer_initial_sync_checkout", default-features = false, features = ["groth16", "parameters", "scale"] }
 manta-primitives = { path = "../../primitives/manta", default-features = false }
-manta-util = { git = "https://github.com/manta-network/manta-rs.git", tag = "v0.5.12", default-features = false }
+manta-util = { git = "https://github.com/manta-network/manta-rs.git", branch = "signer_initial_sync_checkout", default-features = false }
```
Comment (ghzlatarev) [inline pallets/manta-pay/Cargo.toml]: todo: tag

**Comment by zqhxuyuan** [comment]
@SupremoUGH Using manta-rs latest rev commit: f6606ca, has compile issue:
```
error: failed to select a version for `signature`.
    ... required by package `ecdsa v0.14.4`
    ... which satisfies dependency `ecdsa-core = "^0.14"` of package `k256 v0.11.0`
    ... which satisfies dependency `k256 = "^0.11"` of package `bip32 v0.4.0`
    ... which satisfies dependency `bip32 = "^0.4.0"` of package `manta-pay v0.5.12 (https://github.com/manta-network/manta-rs.git?rev=f6606cae7d9d316f9c84cab49fd8cece4a5dc02a#f6606cae)`
    ... which satisfies git dependency `manta-pay` of package `pallet-manta-pay v4.0.4 (/Users/zqh/code/substrates/Manta/pallets/manta-pay)`
    ... which satisfies path dependency `pallet-manta-pay` (locked to 4.0.4) of package `calamari-runtime v4.0.4 (/Users/zqh/code/substrates/Manta/runtime/calamari)`
    ... which satisfies path dependency `calamari-runtime` (locked to 4.0.4) of package `manta v4.0.4 (/Users/zqh/code/substrates/Manta/node)`
versions that meet the requirements `>=1.5, <1.7` are: 1.6.4, 1.6.3, 1.6.2, 1.5.0

all possible versions conflict with previously selected packages.

  previously selected package `signature v1.4.0`
    ... which satisfies dependency `signature = ">=1.3.1, <1.5"` (locked to 1.4.0) of package `ecdsa v0.13.4`
    ... which satisfies dependency `ecdsa-core = "^0.13"` (locked to 0.13.4) of package `k256 v0.10.4`
    ... which satisfies dependency `k256 = "^0.10.4"` (locked to 0.10.4) of package `frame-support v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#ce10b9f2)`
    ... which satisfies git dependency `frame-support` (locked to 4.0.0-dev) of package `calamari-runtime v4.0.4 (/Users/zqh/code/substrates/Manta/runtime/calamari)`
    ... which satisfies path dependency `calamari-runtime` (locked to 4.0.4) of package `manta v4.0.4 (/Users/zqh/code/substrates/Manta/node)`
```
Try to manual set signature version to 1.5.0 on calamari runtime:
```
error: failed to select a version for `signature`.
    ... required by package `calamari-runtime v4.0.4 (/Users/zqh/code/substrates/Manta/runtime/calamari)`
    ... which satisfies path dependency `calamari-runtime` (locked to 4.0.4) of package `manta v4.0.4 (/Users/zqh/code/substrates/Manta/node)`
versions that meet the requirements `^1.5.0` are: 1.6.4, 1.6.3, 1.6.2, 1.5.0

all possible versions conflict with previously selected packages.

  previously selected package `signature v1.3.1`
    ... which satisfies dependency `signature = ">=1.3.1, <1.5"` of package `ecdsa v0.13.3`
    ... which satisfies dependency `ecdsa-core = "^0.13"` of package `k256 v0.10.4`
    ... which satisfies dependency `k256 = "^0.10.4"` of package `frame-support v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#ce10b9f2)`
    ... which satisfies git dependency `frame-support` (locked to 4.0.0-dev) of package `calamari-runtime v4.0.4 (/Users/zqh/code/substrates/Manta/runtime/calamari)`
    ... which satisfies path dependency `calamari-runtime` (locked to 4.0.4) of package `manta v4.0.4 (/Users/zqh/code/substrates/Manta/node)`

failed to select a version for `signature` which could resolve this conflict
```

I think in manta-rs, upgrade `bip32 = "^0.4.0"` only when runtime decide to upgrade to v0.9.37+. current substrate frame-support is using old version of k256.

**Comment by Dengjianping** [comment]
And maybe this new method should be added to zkSBT as well, but we can do it in a separate PR. @zqhxuyuan @ferrell-code

**Comment by zqhxuyuan** [comment]
> And maybe this new method should be added to zkSBT as well, but we can do it in a separate PR. @zqhxuyuan @ferrell-code

As zkSBT is already merge to main branch, so we can add this rpc call into zkSBT pallet together. I already have a seperate branch: [sbt_extension](https://github.com/Manta-Network/Manta/commits/sbt_extension), @SupremoUGH maybe just pick that branch's latest three commit into this branch.

**Review by zqhxuyuan — DISMISSED** [review]

**Review by Dengjianping — DISMISSED** [review]

**Review by zqhxuyuan — APPROVED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**